tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/javainterop/basic/StaticMethodTest.java

```diff
@@ -169,8 +169,8 @@ public void testCreateRawDetails() {
     }
 
     @Test(expectedExceptions = BLangTestException.class,
-          expectedExceptionsMessageRegExp = ".*Invalid update of record field: modification not allowed on readonly " +
-                  "value.*")
+            expectedExceptionsMessageRegExp = "error: \\{ballerina/lang.map}InherentTypeViolation \\{\"message\":" +
+                    "\"cannot update 'readonly' field 'name' in record of type 'Details & readonly'\".*")
     public void testCreateDetails() {
         BRunUtil.invoke(result, "testCreateDetails");
```

Comment:
If the entire record is immutable, the former error is better?

Reply:
The current error message is the one we normally get, when we updating a immuatable value